### ISSUE TITLE
💅 Biome ignores a `biome-ignore` for `useArrowFunction`.

### ISSUE THREAD

**Issue by lgarron** [biomejs/biome#5739]
### Environment information

```bash
See repro.
```

### Rule name

lint/complexity/useArrowFunction

### Playground link

The playground exhibits an unrelated issue with the repro code.

### Expected result

Repro: https://github.com/lgarron/biome-ignore-repro

```
git clone https://github.com/lgarron/biome-ignore-repro && cd biome-ignore-repro
npm install
npx @biomejs/biome check
```

This should not produce an error, as all the `biome-ignore` comments are placed where Biome puts them using "Suppress rule …".

This seems to be a weird parsing edge case, since:

- Removing the unrelated `finallyConstructor(…)` function implementation removes the error.
- Converting the string concatenation to a template (instead of using `biome-ignore` on it) also removes the error.

Note that the code in question is used for a polyfill that is backwards compatible as far back as Internet Explorer 9 and needs to stay that way. Any structural code refactoring would a risk. (That said, I do not want to ignore the given file entirely. It could result in masking other errors, compared to ignoring linter errors explicitly and individually.)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by siketyan** [comment]
Sorry, we have no ETA for a new release now. The first stable version with fix for this will be v2.0.
